gpslogger/src/main/java/com/mendhak/gpslogger/senders/sftp/SFTPJob.java

```diff
@@ -31,6 +33,15 @@
 
     public SFTPJob(File localFile, String remoteDir, String host, int port, String pathToPrivateKey, String privateKeyPassphrase, String username, String password, String hostKey) {
         super(new Params(1).requireNetwork().persist().addTags(getJobTag(localFile)));
+
+        try {
+            Security.removeProvider(BouncyCastleProvider.PROVIDER_NAME);
+            Security.insertProviderAt(new BouncyCastleProvider(), 1);
```

Comment:
BouncyCastle contains the modern algorithms.  By including it this way, the new jsch library will use it when talking to SFTP servers.